index.js

```diff
@@ -43,6 +45,10 @@ async function assumeRole(params) {
   let roleArn = roleToAssume;
   if (!roleArn.startsWith('arn:aws')) {
     // Supports only 'aws' partition. Customers in other partitions ('aws-cn') will need to provide full ARN
+  assert(
+      isDefined(sourceAccountId),
+      "Source Account ID is needed if the Role Name is provided and not the Role Arn."
```

Comment:
Can the user provide this information? I wonder if we can provide a suggested action here if we end up in this error scenario 🤔

Reply:
This is what I was struggling with. It seems like we're trying to help the customer by saying "if you only provide a Role Name, we'll try our best to figure out which account you're talking about". Customers will encounter this exception if they supply a Role Name for use with the OIDC. Our current logic has the STS client call "get-caller-identity" and parse the response to get an account ID. I feel like this is a bit too much magic for customers who may be running this in a self-hosted runner with some limited permissions in AccountA and we might try to accidentally assume another role in accountA. I'd prefer that for the OIDC users we ask them to explicitly provide the Arn and we can relax it later.